### PR TITLE
Corrected modular arithmetic for elliptic curves

### PR DESCRIPTION
**Pull request by alexthed1rk**
#### References to other Issues or PRs

PR related to [#24517](https://github.com/sympy/sympy/issues/24517)

#### Brief description of what is fixed or changed
Replaced all `Eq` usage into `Poly` in `EllipticCurve` class,
so the modular arithmetic applied correctly.
Changed `__repr__` for `EllipticCurve` class accordingly.
Fixed `points_x` for `EllipticCurve` class, so it works for
elliptic curves with `QQ` domain correctly.

#### Release Notes

* ntheory
  * Corrected modular arithmetic for elliptic curves.

### CODE REVIEW

File: sympy/ntheory/elliptic_curve.py [hunk 1]
```diff
@@ -77,10 +76,10 @@ def __contains__(self, point):
             raise ValueError('Invalid point.')
         if self.characteristic == 0 and z1 == 0:
             return True
-        return self._eq.subs({self.x: x1, self.y: y1, self.z: z1})
+        return self._poly.subs({self.x: x1, self.y: y1, self.z: z1}) == 0
 
     def __repr__(self):
-        return 'E({}): {}'.format(self._domain, self._eq)
+        return 'E({}): Eq({}, 0)'.format(self._domain, self._poly.expr)
```
[the remark this answers is not in the document]
Comment (DarkNisa) [reply sympy/ntheory/elliptic_curve.py:82]: I'm open to suggestions. Old representation showed domain and equation
of an elliptic curve, so by moving to polynomials, I decided to leave it as it's.

**Comment by DarkNisa** [comment]
It's my first time making pull requests to such project, so I followed
the guide in documentation. I'll fix doctest failure. I also consider
adding additional tests over elliptic curve class and making another 
pull request.

**Comment by oscarbenjamin** [comment]
It is not necessary to close the pull request to make further changes. You can just push another commit to your branch.

File: sympy/ntheory/elliptic_curve.py [hunk 2]
```diff
@@ -1,23 +1,22 @@
...
-
 class EllipticCurve:
     """
     Create the following Elliptic Curve over domain.
 
     `y^{2} + a_{1} x y + a_{3} y = x^{3} + a_{2} x^{2} + a_{4} x + a_{6}`
 
     The default domain is ``QQ``. If no coefficient ``a1``, ``a2``, ``a3``,
-    it create curve as following form.
+    it creates curve as following form.
```
Comment (smichr) [inline sympy/ntheory/elliptic_curve.py:19]:
```suggestion
    is given then it creates a curve with the following form:
```

File: sympy/ntheory/elliptic_curve.py [hunk 3]
```diff
@@ -132,14 +131,15 @@ def points(self):
             raise ValueError("Infinitely many points")
 
     def points_x(self, x):
-        "Returns points on with curve where xcoordinate = x"
+        """Returns points on with curve where x coordinate = x"""
```
Comment (smichr) [inline sympy/ntheory/elliptic_curve.py:134]:
```suggestion
        """Returns points on with curve where x-coordinate = x"""
```

File: .mailmap [hunk 4]
```diff
@@ -272,6 +272,7 @@ Alexander Dunlap <alexander.dunlap@gmail.com>
 Alexander Eberspächer <alex.eberspaecher@gmail.com>
 Alexander Hirzel <alex@hirzel.us>
 Alexander Pozdneev <pozdneev@users.noreply.github.com>
+Alexander Zhura <nice.zhura@list.ru>
```
Comment (1e9abhi1e10) [inline .mailmap:275]: You need to run the `.mailmap` file by the following command `bin/mailmap_check.py` after adding your name and email.
https://docs.sympy.org/dev/contributing/new-contributors-guide/workflow-process.html#add-your-name-and-email-address-to-the-mailmap-file

Comment (DarkNisa) [reply .mailmap:275]: The problem was that I committed changes in web GitHub and it placed wrong author name. I previously added myself into this list and everything worked. Helpfully with git amend and rebase I fixed such problem and now everything should work. Please check error messages better next time.

File: sympy/ntheory/elliptic_curve.py [hunk 5]
```diff
@@ -132,14 +131,15 @@ def points(self):
             raise ValueError("Infinitely many points")
 
     def points_x(self, x):
-        "Returns points on with curve where xcoordinate = x"
+        """Returns points on with curve where x-coordinate = x"""
```
Comment (smichr) [inline sympy/ntheory/elliptic_curve.py:134]:
```suggestion
        """Returns points on the curve for the given x-coordinate."""
```

**Comment by smichr** [comment]
I am just leaving small text edits. I can't comment on the the rest of the work, however.

**Comment by oscarbenjamin** [comment]
> For whatever reason it's not letting me commit @smichr's suggested docstring fix

Not sure why that is. I just tried and it seemed to work.

**Comment by asmeurer** [comment]
GitHub was having some issues earlier https://www.githubstatus.com/

**Comment by oscarbenjamin** [comment]
Thanks @DarkNisa

**Pull request merged**